tests/audit/report_test.py

```diff
@@ -0,0 +1,128 @@
+import os
+import random
+import string
+import tempfile
+
+import pytest
+
+from detect_secrets.audit.report import generate_report
+from detect_secrets.audit.report import SecretClassToPrint
+from detect_secrets.constants import VerifiedResult
+from detect_secrets.core import baseline
+from detect_secrets.core.secrets_collection import SecretsCollection
+from detect_secrets.plugins.basic_auth import BasicAuthDetector
+from detect_secrets.plugins.jwt import JwtTokenDetector
+from testing.factories import potential_secret_factory as original_potential_secret_factory
+
+
+CREATED_FILES = []
+
+
+@pytest.mark.parametrize(
+    'class_to_print, expected_real, expected_false',
+    [
+        (None, 2, 2),
+        (SecretClassToPrint.REAL_SECRET, 2, 0),
+        (SecretClassToPrint.FALSE_POSITIVE, 0, 3),
+    ],
+)
+def test_generate_report(class_to_print, expected_real, expected_false):
+    filename = baseline_file()
+    output = generate_report(filename, class_to_print)
```

Comment:
Maybe also add a test for format? e.g.

```python
assert output[0] == {...}
```

I think you crafted good test cases to test different situations (e.g. if a secret is marked both as false positive, and true positive, secrets that appear on multiple lines) -- it would be good to assert them too.

On that note, should we also test for cases when `secret.is_secret == None`?